### ISSUE TITLE
Contrast themes – (legacy) Dropdown buttons styled incorrectly in forced colors

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8827]
It looks like in #7544 we changed the styles of our "legacy" dropdowns for WHCM, in a way that only supports the default dark theme. It’s not correct for us to apply `forced-color-adjust: none` and custom "white on black" styles like was done there – instead, we need to style the dropdown with system colors.

Here is the component in question in our styleguide, in forced colors "light" to the left and without forced colors to the right:

![Image](https://user-images.githubusercontent.com/877585/178235701-7a347eb9-c472-486a-8b9c-86002bfa35df.png)

Figma link: https://www.figma.com/file/O81E48wieOZ7e6AivvksWn/Contrast-themes%3A-Wagtail-audit?node-id=6766%3A4147

---

Implementation: https://github.com/wagtail/wagtail/blob/main/client/scss/components/_dropdown.legacy.scss

In particular: https://github.com/wagtail/wagtail/blob/7286c530e97e2eb64f3747c6904a7456789f82b7/client/scss/components/_dropdown.legacy.scss#L53-L78
